underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -120,7 +120,9 @@ private void openStream() {
       return;
     }
     GetObjectRequest getReq = new GetObjectRequest(mBucketName, mKey);
-    getReq.setRange(mPos);
+    if (mPos > 0) {
+      getReq.setRange(mPos);
```

Comment:
Leaving the range unset is equivalent to `setRange(0)`, since that means to fetch the entire file. But this breaks for a 0 length file. I've added a comment.